peewee_async/pool.py

```diff
@@ -68,14 +78,74 @@ async def create(self) -> None:
         """
         if "connect_timeout" in self.connect_params:
             self.connect_params['timeout'] = self.connect_params.pop("connect_timeout")
-        self.pool = cast(
-            PoolProtocol,
-            await aiopg.create_pool(
-                database=self.database,
-                **self.connect_params
-            )
+        self.pool = await aiopg.create_pool(
+            database=self.database,
+            **self.connect_params
+        )
+
+
+class PsycopgPoolBackend(PoolBackend):
+    """Asynchronous database connection pool based on psycopg + psycopg_pool libraries.
+    """
+
+    async def create(self) -> None:
+        """Create connection pool asynchronously.
+        """
+        params = self.connect_params.copy()
+        pool = psycopg_pool.AsyncConnectionPool(
+            format_dsn(
+                'postgresql',
+                host=params.pop('host'),
+                port=params.pop('port'),
+                user=params.pop('user'),
+                password=params.pop('password'),
+                path=self.database,
+            ),
+            kwargs={
+                'cursor_factory': psycopg.AsyncClientCursor,
+                'autocommit': True,
+            },
+            **params,
         )
 
+        await pool.open()
+        self.pool = pool
+
+    def has_acquired_connections(self) -> bool:
+        if self.pool is not None:
+            return bool(self.pool._nconns - self.pool._num_pool > 0)
```

Comment:
**_num_pool** is used for generating pool name it shouldn't be here. I proposed there should be the line:
```python return bool(self.pool._nconns - len(self.pool._pool) > 0)``` 
But got an error on t**est_transaction_error_on_begin** test. It needs to investigate.

Reply:
Found that a small pause is needed, because the pool simply does not have time to initialize before min_size connections, so the test ends with an error